quinn-udp/src/windows.rs

```diff
@@ -68,38 +133,162 @@ impl UdpSocketState {
             if rc == -1 {
                 return Err(io::Error::last_os_error());
             }
+
+            if wsa_recvmsg_ptr.is_some() {
```

Comment:
How do the below options interact with IPv4 packets received on dual-stack sockets? On Linux we have to set the IPv4 option `IP_RECVTOS` (probably analogous to Windows's `IP_ECN`) for dual-stack sockets to get ECN information for IPv4 packets.

Reply:
Nice catch, i will do some test to see how it works

Reply:
I had no time to test this, the code is different from the unix side and i dunno how windows work with dual stack sockets. Does quinn has a dual stack regression tests ?

Reply:
We have basic end-to-end tests, but those won't detect missing metadata. Windows IPv6 sockets are not dual-stack by default but may be configured as such. We should probably add some quinn-udp unit tests for this.

I think we can consider this non-blocking, because the status quo doesn't support these features *ever*.